packages/inventory-compliance/src/SystemRulesTable.js

```diff
@@ -166,22 +168,46 @@ class SystemRulesTable extends React.Component {
         ));
     }
 
+    rowRefId = (row) => {
+        const { columnIndices, refIds } = this.state;
+        return refIds[row.cells[columnIndices.TITLE_COLUMN].original];
+    }
+
     onSelect = (_event, selected, key) => {
+        const { remediationsEnabled, tailoringEnabled, handleSelect } = this.props;
         const { columnIndices, page, itemsPerPage } = this.state;
-        let { currentRows, rows } = this.state;
+        let { currentRows, rows, selectedRefIds } = this.state;
+
         if (key === -1) {
             rows = this.selectAll(rows, selected);
             currentRows = this.selectAll(currentRows, selected);
+            if (selected) {
+                selectedRefIds = rows.filter(row => this.isParent(row)).map(row => this.rowRefId(row));
+            } else {
+                selectedRefIds = [];
+            }
         } else {
             // One rule was selected
             const index = ((page - 1) * itemsPerPage * 2) + Number(key);
-            if (currentRows[index].cells[columnIndices.REMEDIATIONS_COLUMN].original) {
+            if (tailoringEnabled || remediationsEnabled && currentRows[index].cells[columnIndices.REMEDIATIONS_COLUMN].original) {
                 rows[index].selected = selected;
                 currentRows[key].selected = selected;
+
+                const currentRefId = this.rowRefId(rows[index]);
+                const currentRefIdInSelectedRefIds = selectedRefIds.indexOf(currentRefId);
+                if (selected && currentRefIdInSelectedRefIds === -1) {
+                    selectedRefIds.push(currentRefId);
+                } else if (!selected && currentRefIdInSelectedRefIds !== -1) {
+                    selectedRefIds.splice(currentRefIdInSelectedRefIds, 1);
+                }
             }
         }
 
-        this.setState({ rows, currentRows });
+        this.setState({
+            rows,
+            currentRows,
+            selectedRefIds
+        }, () => handleSelect(selectedRefIds));
```

Comment:
This `handleSelect` is a callback which sends this list to redux-form through https://github.com/RedHatInsights/compliance-frontend/pull/413 - it could serve for any other purpose in the future though